cfme/tests/middleware/test_middleware_deployment.py

```diff
@@ -1,33 +1,48 @@
-from random import sample
-
 import pytest
+from cfme.middleware import get_random_list
 from cfme.middleware.deployment import MiddlewareDeployment
+from cfme.middleware.server import MiddlewareServer
 from utils import testgen
 from utils.version import current_version
 
 pytestmark = [
-    pytest.mark.usefixtures('single_middleware_provider'),
+    pytest.mark.usefixtures('setup_provider'),
```

Comment:
Added code to get `provider`'s object with `get_crud(get_provider_key(provider_name))`.
* For deployment page on `MiddlewareDeployment.deployments()` we will not get `provider` detail. This page does not have provider data on list all deployments.
* Added code to get deployments data from `mgmt` system